File: DCP-1838/solution.swift
// Problem 1838 (count smaller elements on the right)

import Foundation

func countSmallerElements(_ nums: [Int]) -> [Int] {
    var result = [Int]()
    
    for i in 0..<nums.count {
        var count = 0
        for j in (i + 1)..<nums.count {
            if nums[j] < nums[i] {
                count += 1
            }
        }
        result.append(count)
    }
    return result
}

File: DCP-1838/solution.kt
// Problem 1838

fun countSmallerElements(nums: IntArray): IntArray {
    val result = IntArray(nums.size)
    
    for (i in nums.indices) {
        var count = 0
        for (j in (i + 1) until nums.size) {
            if (nums[j] < nums[i]) {
                count++
            }
        }
        result[i] = count
    }
    return result
}